aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -254,122 +254,144 @@ pub(crate) fn validate_authenticators(
+            &training_wheels_pk,
+            config,
+            pvk,
+        )?;
+    }
 
-        match &sig.cert {
-            EphemeralCertificate::ZeroKnowledgeSig(zksig) => match jwk {
-                JWK::RSA(rsa_jwk) => {
-                    if zksig.exp_horizon_secs > config.max_exp_horizon_secs {
-                        // println!("[aptos-vm][groth16] Expiration horizon is too long");
-                        return Err(invalid_signature!("The expiration horizon is too long"));
-                    }
-
-                    // If an `aud` override was set for account recovery purposes, check that it is
-                    // in the allow-list on-chain.
-                    if zksig.override_aud_val.is_some() {
-                        config.is_allowed_override_aud(zksig.override_aud_val.as_ref().unwrap())?;
-                    }
-
-                    match &zksig.proof {
-                        ZKP::Groth16(groth16proof) => {
-                            // let start = std::time::Instant::now();
-                            let public_inputs_hash = get_public_inputs_hash(
-                                sig,
-                                pk.inner_keyless_pk(),
-                                &rsa_jwk,
-                                config,
-                            )
-                            .map_err(|_| {
-                                // println!("[aptos-vm][groth16] PIH computation failed");
-                                invalid_signature!("Could not compute public inputs hash")
-                            })?;
-                            // println!("Public inputs hash time: {:?}", start.elapsed());
-
-                            let groth16_and_stmt =
-                                Groth16ProofAndStatement::new(*groth16proof, public_inputs_hash);
-
-                            // The training wheels signature is only checked if a training wheels PK is set on chain
-                            if training_wheels_pk.is_some() {
-                                match &zksig.training_wheels_signature {
-                                    Some(training_wheels_sig) => {
-                                        training_wheels_sig
-                                            .verify(
-                                                &groth16_and_stmt,
-                                                training_wheels_pk.as_ref().unwrap(),
+    Ok(())
+}
+
+pub fn verify_keyless_signature_without_ephemeral_signature_check(
+    public_key: &AnyKeylessPublicKey,
+    signature: &KeylessSignature,
+    jwk: &JWK,
+    exp_date_secs: u64,
+    training_wheels_pk: &Option<EphemeralPublicKey>,
+    config: &Configuration,
+    pvk: &Option<PreparedVerifyingKey<Bn254>>,
+) -> Result<(), VMStatus> {
+    signature.verify_expiry(exp_date_secs).map_err(|_| {
+        // println!("[aptos-vm][groth16] ZKP expired");
+
+        invalid_signature!("The ephemeral keypair has expired")
+    })?;
```

Comment:
this seems new logic, feature-gate it?

Reply:
oh do you really need it? the check is done already in L208 ?

Reply:
Remove that check from `validate_authenticators` then! 🙏  (assuming it is safe and does not introduce a DoS; there was a reason we were doing it early)

Reply:
Or, leave it there and add a comment: 

```
// This is a redundant check to quickly dismiss expired signatures early and mitigate against DoS attacks. The actual check is performed in `verify_keyless_signature_without_ephemeral_signature_check`.

Reply:
done